src/utils/errorSerializer.ts

```diff
@@ -34,60 +38,81 @@ import UserTimeoutError from "../errors/UserTimeoutError";
 import VersionInvalidError from "../errors/VersionInvalidError";
 import VersionMismatchError from "../errors/VersionMismatchError";
 
-const errorCodeToErrorClass: {[key: string]: any} = {
+const errorCodeToErrorClass: Record<keyof typeof ErrorCode, KnownErrorConstructors> = {
   [ErrorCode.ERR_WEBEID_ACTION_PENDING]:          ActionPendingError,
   [ErrorCode.ERR_WEBEID_ACTION_TIMEOUT]:          ActionTimeoutError,
   [ErrorCode.ERR_WEBEID_CONTEXT_INSECURE]:        ContextInsecureError,
   [ErrorCode.ERR_WEBEID_EXTENSION_UNAVAILABLE]:   ExtensionUnavailableError,
+  [ErrorCode.ERR_WEBEID_MISSING_PARAMETER]:       MissingParameterError,
   [ErrorCode.ERR_WEBEID_NATIVE_INVALID_ARGUMENT]: NativeInvalidArgumentError,
   [ErrorCode.ERR_WEBEID_NATIVE_FATAL]:            NativeFatalError,
   [ErrorCode.ERR_WEBEID_NATIVE_UNAVAILABLE]:      NativeUnavailableError,
   [ErrorCode.ERR_WEBEID_USER_CANCELLED]:          UserCancelledError,
   [ErrorCode.ERR_WEBEID_USER_TIMEOUT]:            UserTimeoutError,
   [ErrorCode.ERR_WEBEID_VERSION_INVALID]:         VersionInvalidError,
   [ErrorCode.ERR_WEBEID_VERSION_MISMATCH]:        VersionMismatchError,
+  [ErrorCode.ERR_WEBEID_UNKNOWN_ERROR]:           UnknownError,
 };
 
-export function serializeError(error: any): any {
-  const {
-    message,
-    name,
-    fileName,
-    lineNumber,
-    columnNumber,
-    stack,
-  } = error;
-
-  return {
-    ...(
-      Object.fromEntries(
-        Object.getOwnPropertyNames(error)
-          .map((prop) => [prop, error[prop]])
-      )
-    ),
-
-    message,
-    name,
-    fileName,
-    lineNumber,
-    columnNumber,
-    stack,
-  };
+export function serializeError(error: Error): SerializedError {
+  if (typeof error != "object" || !(error instanceof Error)) {
```

Comment:
The `typeof` check looks redundant as no non-object can be an instance of `Error`:
```suggestion
  if (!(error instanceof Error)) {
```